src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -2986,6 +2986,103 @@ public async Task TestFrozenPartialProjectHasDifferentSemanticVersions()
                 await frozenDocument.Project.GetSemanticVersionAsync());
         }
 
+        [Fact]
+        public async Task TestFrozenPartialProjectHasDifferentSemanticVersions_ChangedDoc1()
+        {
+            using var workspace = WorkspaceTestUtilities.CreateWorkspaceWithPartialSemantics();
+            var project = workspace.CurrentSolution.AddProject("CSharpProject", "CSharpProject", LanguageNames.CSharp);
+            project = project.AddDocument("Extra.cs", SourceText.From("class Extra { }")).Project;
+
+            var documentToFreezeOriginal = project.AddDocument("DocumentToFreeze.cs", SourceText.From("class DocumentToFreeze { void M() { } }"));
+            project = documentToFreezeOriginal.Project;
+            var compilation = await project.GetCompilationAsync();
+
+            var solution = project.Solution.WithDocumentText(documentToFreezeOriginal.Id, SourceText.From("class DocumentToFreeze { void M() { /*no top level change*/ } }"));
+            var documentToFreezeChanged = solution.GetDocument(documentToFreezeOriginal.Id);
+            var tree = await documentToFreezeChanged.GetSyntaxTreeAsync();
+
+            var frozenDocument = documentToFreezeChanged.WithFrozenPartialSemantics(CancellationToken.None);
+
+            // Because we had no compilation produced yet, we expect that only the DocumentToFreeze is in the compilation
+            Assert.NotSame(frozenDocument, documentToFreezeChanged);
+
+            // Versions should the same since there wasn't a top level change different
+            Assert.Equal(
+                await documentToFreezeOriginal.GetTopLevelChangeTextVersionAsync(),
+                await frozenDocument.GetTopLevelChangeTextVersionAsync());
+
+            Assert.Equal(
+                await documentToFreezeChanged.GetTopLevelChangeTextVersionAsync(),
+                await frozenDocument.GetTopLevelChangeTextVersionAsync());
+
+            Assert.Equal(
+                await documentToFreezeOriginal.Project.GetDependentSemanticVersionAsync(),
+                await frozenDocument.Project.GetDependentSemanticVersionAsync());
+
+            Assert.Equal(
+                await documentToFreezeChanged.Project.GetDependentSemanticVersionAsync(),
+                await frozenDocument.Project.GetDependentSemanticVersionAsync());
+
+            Assert.Equal(
+                await documentToFreezeOriginal.Project.GetSemanticVersionAsync(),
+                await frozenDocument.Project.GetSemanticVersionAsync());
+
+            Assert.Equal(
+                await documentToFreezeChanged.Project.GetSemanticVersionAsync(),
+                await frozenDocument.Project.GetSemanticVersionAsync());
+
+            GC.KeepAlive(tree);
```

Comment:
Is this still needed given trees are now strongly held? I could understand if this was copy/paste from a previous test but is now obsolete.

Reply:
oh, you're right.  i can remove.  thanks!